libraries/redcore/api/hal/hal.php

```diff
@@ -662,13 +669,14 @@ public function apiCreate()
 		}
 
 		$this->setData('result', $result);
-		$this->triggerFunction('displayErrors', $model);
+		$this->triggerFunction('displayErrors', $model->getErrors());
 
 		if ($this->statusCode < 400)
 		{
-			if ($result === false)
+			if ($result === false || !empty($model->getErrors()))
```

Comment:
you cannot be sure if the model has getErrors method, but why are you even checking it here? it is not fatal errors at this point or the result would be false